From 34e26017e3c2432a6675e5ce6a1c64163a05e4fb Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 14 Nov 2016 14:19:58 -0500
Subject: [PATCH] appchooser: Plug a memory leak

We were not dropping the reference that the model gives us.

https://bugzilla.gnome.org/show_bug.cgi?id=774352
---
 gtk/gtkappchooserwidget.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gtk/gtkappchooserwidget.c b/gtk/gtkappchooserwidget.c
index f45348e735..9a4103a8fc 100644
--- a/gtk/gtkappchooserwidget.c
+++ b/gtk/gtkappchooserwidget.c
@@ -161,17 +161,17 @@ refresh_and_emit_app_selected (GtkAppChooserWidget *self,
       if (!g_app_info_equal (self->priv->selected_app_info, info))
         {
           should_emit = TRUE;
-          g_object_unref (self->priv->selected_app_info);
-
-          self->priv->selected_app_info = info;
+          g_set_object (&self->priv->selected_app_info, info);
         }
     }
   else
     {
       should_emit = TRUE;
-      self->priv->selected_app_info = info;
+      g_set_object (&self->priv->selected_app_info, info);
     }
 
+  g_object_unref (info);
+
   if (should_emit)
     g_signal_emit (self, signals[SIGNAL_APPLICATION_SELECTED], 0,
                    self->priv->selected_app_info);
